engine/access/rest/websockets/controller.go

```diff
@@ -4,100 +4,225 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"sync"
+	"time"
 
 	"github.com/google/uuid"
 	"github.com/gorilla/websocket"
 	"github.com/rs/zerolog"
 
 	dp "github.com/onflow/flow-go/engine/access/rest/websockets/data_provider"
 	"github.com/onflow/flow-go/engine/access/rest/websockets/models"
-	"github.com/onflow/flow-go/engine/access/state_stream"
-	"github.com/onflow/flow-go/engine/access/state_stream/backend"
 	"github.com/onflow/flow-go/utils/concurrentmap"
 )
 
+const (
+	// PingPeriod defines the interval at which ping messages are sent to the client.
+	// This value must be less than pongWait.
```

Comment:
I guess Ulyana took it from here https://github.com/gorilla/websocket/blob/v1.5.3/examples/chat/client.go#L23.
I believe it's because ping and pong share the same timer.

Let’s consider a case where pongWait is smaller than pingPeriod, and we’ll see why this configuration is problematic.

**Parameters:**
pongWait = 30s
pingPeriod = 40s

**At t=0:**
The server sends a ping message to the client.

**At t=30s:**
The pongWait expires because the server hasn't received a pong (or any message) from the client.
The server assumes the connection is dead and closes it.

**At t=40s:**
The server sends its second ping, but the connection is already closed due to the timeout at t=30s.

Reply:
yea, but in that case, the server should have cleaned up the ping service when the connection was closed, so the second ping would never happen

Reply:
The server uses `pong` responses to confirm that the client is still alive and able to communicate. If the `pong` isn't received within the `PongWait` period, the server assumes the client is unresponsive and closes the connection.

By setting `PingPeriod` to a smaller value than `PongWait`, the server ensures it sends a ping well before the `PongWait` timeout elapses. Each new `pong` message resets the server's read deadline, keeping the connection alive as long as the client is responsive.

Configuration:
    `PongWait` = 10 seconds
    `PingPeriod` = 9 seconds (90% of `PongWait`)

Example:
At t=9, the server sends a ping
At t=10, the client responds with a pong. The server resets its read deadline to t=20.
At t=18, the server sends another ping. If the client responds with a pong at t=19, the read deadline is extended to t=29.

In case of failure:
If the client stops responding, the server will send a ping at t=9 but won't receive a pong by t=10. The server then closes the connection.
	
    
 If `PingPeriod` were greater than or equal to `PongWait`, the server could fail to send a `ping` message in time to receive a `pong` before the timeout expires. This would cause the server to terminate connection unnecessarily.